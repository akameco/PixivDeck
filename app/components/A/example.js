// @flow
import * as React from 'react'
import { storiesOf } from '@storybook/react'
import A from '.'

storiesOf('A', module).add('basic', () => {
  return (
    <div>
      Hello
      <A>Hello</A>
      Hello
    </div>
  )
})
