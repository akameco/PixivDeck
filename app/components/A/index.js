// @flow
import styled from 'styled-components'

const A = styled.a`
	color: #5d89a8;

	&:hover {
		color: #6cc0e5;
	}
`

export default A
