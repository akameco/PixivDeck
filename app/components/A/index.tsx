import styled from 'styled-components'
import { key } from 'styles/styleTheme'

const A = styled.a`
  color: ${key('blue')};
  cursor: pointer;

  &:hover {
    text-decoration: underline;
  }
`
export default A
