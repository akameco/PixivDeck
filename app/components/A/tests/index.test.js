import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import A from '../'

const href = 'https://akameco.github.io'
const children = <h1>Test</h1>

const renderComponent = (props = {}) =>
  shallow(
    <A href={href} {...props}>
      {children}
    </A>
  )

test('render an <a> tag', () => {
  const c = renderComponent()
  expect(c.type()).toEqual('a')
})

test('have an href attribute', () => {
  const c = renderComponent()
  expect(c.prop('href')).toEqual(href)
})

test('have children', () => {
  const c = renderComponent()
  expect(c.contains(children)).toEqual(true)
})

test('have a className attribute', () => {
  const className = 'test'
  const c = renderComponent({ className })
  expect(c.find('a').hasClass(className)).toEqual(true)
})

test('adopt a target', () => {
  const target = '_blank'
  const c = renderComponent({ target })
  expect(c.prop('target')).toEqual(target)
})

test('adopt a type attribute', () => {
  const type = 'text/html'
  const c = renderComponent({ type })
  expect(c.prop('type')).toEqual(type)
})

test('snapshot', () => {
  const c = renderComponent()
  expect(shallowToJson(c)).toMatchSnapshot()
})
