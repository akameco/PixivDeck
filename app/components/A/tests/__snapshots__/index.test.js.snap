exports[`test snapshot 1`] = `
<a
  className="svQEm"
  href="https://akameco.github.io">
  <h1>
    Test
  </h1>
</a>
`;
